### PR TITLE
fix(shapes): fabric.Object._fromObject never should return

### PR DESCRIPTION
**Pull request by rockerBOO**
`fabric.Object._fromObject` doesn't return anything, and is processed through the callback.

This removes the return and fixes the documentation. This is mostly to upstream this fix to Typescript types as it incorrectly thinks Polygon, Circle, Ellipse return an object.

Thank you!

### CODE REVIEW

File: src/shapes/polygon.class.js [hunk 1]
```diff
@@ -74,7 +74,7 @@
    * @param {Function} [callback] Callback to invoke when an fabric.Path instance is created
    */
   fabric.Polygon.fromObject = function(object, callback) {
-    return fabric.Object._fromObject('Polygon', object, callback, 'points');
+    fabric.Object._fromObject('Polygon', object, callback, 'points');
```
Comment (asturur) [inline src/shapes/polygon.class.js:77]: can you add return void to this one too?

**Comment by asturur** [comment]
Hi @rockerBOO thanks this is good. Can you address the comment?

File: src/shapes/polygon.class.js (hunk 1 above)
Comment (rockerBOO) [reply src/shapes/polygon.class.js:78]: Added

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
Thank you for the contribution

**Pull request merged**